src/drivers/pca9685_pwm_out/module.yaml

```diff
@@ -1,11 +1,28 @@
 module_name: PCA9685 Output
 actuator_output:
   output_groups:
-    - param_prefix: PCA9685
-      channel_label: 'Channel'
+    - generator: pwm
+      param_prefix: PCA9685
+      channel_labels: ['Channel']
+      channel_label_module_name_prefix: true
+      timer_config_file: "src/drivers/pca9685_pwm_out/timer_config.cpp"
       standard_params:
         disarmed: { min: 800, max: 2200, default: 900 }
         min: { min: 800, max: 1400, default: 1000 }
         max: { min: 1600, max: 2200, default: 2000 }
         failsafe: { min: 800, max: 2200 }
       num_channels: 16
+      pwm_timer_param:
```

Comment:
Instead of using the pwm generator, rather use a per-channel parameter in the form of:
```
          extra_params:
              - name: 'PCA9685${i}_FREQ'
                label: 'Frequency'
                index_offset: 0
                advanced: false
```
And then implement it here: https://github.com/PX4/PX4-Autopilot/blob/master/Tools/module_config/generate_actuators_metadata.py#L267=

Reply:
Hi, I'm back working that now... 

PCA9685 has 16 configurable channels with independent conter, but they share the same timer config, which works kinda like what `pwm` generator would do. But the problem is that unlike those timer configurations on STM32 based chips which needs to be parsed from codes with great flexibility, PCA9685 does only require one simple clk source to be configured, which can be done inside `module.yaml`.

Using per-channel definition on those channels will significantly introduce 16x same cfg, which is quite annoying, I think. The problem is `pwm` generator requires `timer_config_file` to be exist, which is not desired in my case:

```python
if 'timer_config_file' in group:
    timer_config_file = os.path.join(root_dir, group['timer_config_file'])
if timer_config_file is None:
    raise Exception('trying to generate pwm outputs, but --timer-config not set')
```

I wonder if I can add a new case here to implement the ability to replace `get_timer_groups()` call in case of `timer_config_file` missing but similar definition is already available directly inside `module.yaml`, which in theory supports multiple timer groups as well and also has no collision with those on chip PWM functionality

Reply:
Ah there's only a single parameter. Then you can add it like this: https://github.com/PX4/PX4-Autopilot/blob/main/src/drivers/uavcan/module.yaml#L8.